### ISSUE TITLE
[fund / open] allow setting feerate beyond decimal point. + add utxo selection to [open]

### ISSUE THREAD

**Issue by babikpatient6** [alexbosworth/balanceofsatoshis#482]
1)
Problem: In recent mempool there was 60mb+ waiting in the range 1.00-1.02sat/vbyte. By paying >=1.03sat/vbyte I could place my tx on top of 60mb+ in priority.. 
With larger txs this is worthy if someone is efficiency-obsessed as me. Currently BOS wont really respect --fee-rate with any other than natural number (at least as far as I've tried)

Solution: Allow specifying feerate up to 2nd decimal place.

2)
Adding --select-utxos option to [open] as is in [fund] would simplify things..

**Comment by alexbosworth** [comment]
if you use external funding you could choose any fee rate on that wallet

**Comment by babikpatient6** [comment]
> if you use external funding you could choose any fee rate on that wallet

that is true, however since lnd adopted p2tr addresses (cannot be disabled 100%), I havent been able to do this anymore as electrum doesnt support p2tr yet and in other wallets I havent been able to combine utxo from P2TR and P2WPKH together and/or havent been able to select coins the way I wanted.. Any other ideas?

**Comment by alexbosworth** [comment]
OK I see what you mean

you can also select utxos with fund and so you can fund open with internal funds but selecting utxos

**Comment by M1ch43lV** [comment]
Please implement the possibility to allow fractional fund fee rates when using internal funds.
e.g. bos open <pubkey> --internal-fund-at-fee-rate 2.1

**Comment by alexbosworth** [comment]
lnd doesn't internally allow for fractional fee rates so waiting on that change

**Comment by babikpatient6** [comment]
Adding --select-utxos option to [open] with flag including unconfirmed utxos would indeed be very useful.

Afaik --select-utxos doesn't include unconfirmed utxos. Having them there would be also incredibly useful, but they should be somehow highlighted or separated from confirmed utxos, because unconfirmed and even unpublished anchor sweeps regularly cause terrible mess and confusion. Even LND (or RTL rather?) is dumb and will include unpublished 1s/vB anchor sweeps into TX creation if "use unconfirmed outputs" is selected in RTL..

Bumping unconfirmed TXs and selecting only desirable unconfirmed utxos while opening channels is possible with [open] + external wallet, however I often see this being too difficult for lazy noobs. I can spend half an hour explaining them steps and in the end they say thanks and use RTL to cpfp unconfirmed opening tx and then they proceed to open several channels one by one in tandem with RTL anyway..

Option to select utxos and especially unconfirmed utxos direcly in [open] with user able to easily set final effective fee rate rather then "child" fee (relying on user to calculate it in advance with CPFP formula or eyeball and over/underpay) would reduce onchain clutter and friction for noobs by order or magnitude.

**Comment by M1ch43lV** [comment]
I also request this feature of selecting uncofirmed tx with --select-utxos if possible. Sometimes I close multiple channels at very low fee and use all unconfirmed utxos to open a new bigger one.

**Comment by alexbosworth** [comment]
You should be able to select unconfirmed utxos if you specify them manually in fund

**Comment by M1ch43lV** [comment]
I know  :-)  but it's a bit uncomfortable to select all the unconfirmed ones and to construct a string like: 
```bos fund <addr> <amount> --fee-rate xy s/vb --utxo <outpoint1> --utxo <outpoint2> --utxo <outpoint3>```

**Comment by alexbosworth** [comment]
It's by design that it requires more effort because spending unconfirmed UTXOs is a way that people can get into trouble with their funds or compromise privacy, ideally it's not necessary or common

Open allowing for selecting confirmed utxos directly though could be a good idea

**Comment by babikpatient6** [comment]
> Afaik --select-utxos doesn't include unconfirmed utxos.

Ok, sorry for partial missinformation. I was under the impression bos fund can't do that if it doesn't include unconfirmed coins in it's interactive window and it didn't crossed my mind to specify them as outpoints.. I get the point why is it by design, but still including them in window, but somehow separated (maybe behind second y/n interactive window with disclaimer?) would be ideal. In any case allowing both options (--select-utxos and --utxo) directly in open would already reduce friction a lot.

**Comment by mutatrum** [comment]
LND issue to track: https://github.com/lightningnetwork/lnd/issues/7456
